dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/datafile/FileService.java

```diff
@@ -204,4 +216,46 @@ public String scan(InputStream fileInput) throws IOException {
 
     }
 
+    public String checkFilesIntegrity() {
+        List<DataFile> suspicious = new ArrayList<DataFile>();
+        List<DataFile> dataFiles = dataFileService.findAll();
+
+        if (dataFiles.isEmpty()) {
+            return "No datafiles found - check database.";
+        } else {
+
+            int filesCount = dataFiles.size();
+            String message = "Found " + filesCount + " files in repository.";
+            for (DataFile dataFile:dataFiles) {
+                try {
+                    StorageIO<DataFile> storageIO = DataAccess.dataAccess().getStorageIO(dataFile);
+                    if (!storageIO.exists() && storageIO.getSize() == dataFile.getFilesize()) {
```

Comment:
Can you point how `storageIO.getSize()` is populated. I can only find reference to setting this value when `open()` is executed. In other cases seems to me like it's always 0

Reply:
I admit I was a bit messy - now should work properly